From: Michael Catanzaro <mcatanzaro@igalia.com>
Date: Sun, 24 Jun 2018 19:46:19 -0500
Subject: cookie-jar: bail if hostname is an empty string (CVE-2018-12910)

There are several other ways to fix the problem with this function, but
skipping over all of the code is probably the simplest.

Fixes #3

(cherry picked from commit db2b0d5809d5f8226d47312b40992cadbcde439f)

Origin: https://gitlab.gnome.org/GNOME/libsoup/commit/db2b0d5809d5f8226d47312b40992cadbcde439f
Applied-Upstream: 2.62.3
---
 libsoup/soup-cookie-jar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libsoup/soup-cookie-jar.c
+++ b/libsoup/soup-cookie-jar.c
@@ -306,7 +306,7 @@
 
 	priv = SOUP_COOKIE_JAR_GET_PRIVATE (jar);
 
-	if (!uri->host)
+	if (!uri->host || !uri->host[0])
 		return NULL;
 
 	/* The logic here is a little weird, but the plan is that if
